src/github.com/couchbase/sync_gateway/base/bucket.go

```diff
@@ -350,7 +354,7 @@ func GetCouchbaseBucket(spec BucketSpec) (bucket Bucket, err error) {
 	return
 }
 
-func GetBucket(spec BucketSpec) (bucket Bucket, err error) {
+func GetBucket(spec BucketSpec, callback func(bucket string, err error)) (bucket Bucket, err error) {
```

Comment:
What do you think about defining the callback as part of the spec instead of as a separate parameter?